src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Filters/ValidateResourceIdFilterTests.cs

```diff
@@ -75,6 +78,9 @@ public void GivenAnObservationAction_WhenPuttingAnObservationObjectWithoutAnId_T
             var context = CreateContext(observation, Guid.NewGuid().ToString());
 
             Assert.Throws<ResourceNotValidException>(() => filter.OnActionExecuting(context));
+
+            var exception = Assert.Throws<ResourceNotValidException>(() => filter.OnActionExecuting(context));
```

Comment:
You should be able to remove the `Assert.Throws<ResourceNotValidException>(() => filter.OnActionExecuting(context));` above since we're doing the same thing in this line and keeping the exception.